script/spec-runner.js

```diff
@@ -230,8 +230,13 @@ async function runMainProcessElectronTests () {
 }
 
 async function installSpecModules (dir) {
+  // v8 headers use c++17 so override the gyp default of -std=c++14,
+  // but don't clobber any other CXXFLAGS that were passed into spec-runner.js
+  const CXXFLAGS = ['-std=c++17', process.env.CXXFLAGS].filter(x => !!x).join(' ');
```

Comment:
hm... do we need to give this the [nan-spec](https://github.com/electron/electron/blob/d67532ee9f836857bc43989fbb8376d1fe769a48/script/nan-spec-runner.js#L34-L53) treatment?

Reply:
Yes, great catch! I've filed a an upgrades followup and self-assigned @ https://github.com/electron/electron/issues/34209